pkg/process/metadata/parser/java/spring.go

```diff
@@ -0,0 +1,205 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package javaparser contains functions to autodetect service name for java applications
+package javaparser
+
+import (
+	"archive/zip"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/rickar/props"
+	"github.com/vibrantbyte/go-antpath/antpath"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	defaultLocations       = "optional:classpath:/;optional:classpath:/config/;optional:file:./;optional:file:./config/;optional:file:./config/*/"
+	defaultConfigName      = "application"
+	locationPropName       = "spring.config.locations"
+	configPropName         = "spring.config.name"
+	activeProfilesPropName = "spring.profiles.active"
+
+	appnamePropName = "spring.application.name"
+)
+
+// parseURI parses locations (usually specified by the property locationPropName) given the list of active profiles (specified by activeProfilesPropName)
+// and the current directory cwd.
+// It returns a couple of maps each having as key the profile name ("" stands for default one) and as value the ant patterns where the properties should be found
+// The first map returned is the locations to be found in fs while the second map contains locations on the classpath (usually inside the application jar)
+func parseURI(locations []string, name string, profiles []string, cwd string) (map[string][]string, map[string][]string) {
+	classpaths := make(map[string][]string)
+	files := make(map[string][]string)
+	for _, current := range locations {
+		parts := strings.Split(current, ":")
+		pl := len(parts)
+
+		isClasspath := false
+		if pl > 1 && parts[pl-2] == "classpath" {
+			parts[pl-1] = "BOOT-INF/classes/" + parts[pl-1]
+			isClasspath = true
+		}
+
+		doAppend := func(name string, profile string) {
+			name = filepath.Clean(name)
+			if isClasspath {
+				classpaths[profile] = append(classpaths[profile], filepath.ToSlash(name))
+			} else {
+				files[profile] = append(files[profile], filepath.ToSlash(abs(name, cwd)))
+			}
+		}
```

Comment:
I'd extract the method declaration (and allocation) outside of the for-loop.
the best would be to declare it as a normal function (instead of a sub-method)